Source/WebCore/style/StyleBuilderCustom.h

```diff
@@ -1258,13 +1258,23 @@ inline void BuilderCustom::applyValueContent(BuilderState& builderState, CSSValu
             builderState.style().setHasAttrContent();
         else
             const_cast<RenderStyle&>(builderState.parentStyle()).setHasAttrContent();
-        QualifiedName attr(nullAtom(), primitiveValue.stringValue().impl(), nullAtom());
+
+        auto value = primitiveValue.cssAttrValue();
+        QualifiedName attr(nullAtom(), value->attributeName().impl(), nullAtom());
         const AtomString& attributeValue = builderState.element() ? builderState.element()->getAttribute(attr) : nullAtom();
 
         // Register the fact that the attribute value affects the style.
         builderState.registerContentAttribute(attr.localName());
 
-        return attributeValue.isNull() ? emptyAtom() : attributeValue.impl();
+        if (attributeValue.isNull()) {
+            auto fallback = value->fallback();
+            if (fallback != nullptr) {
+                if (fallback->isPrimitiveValue() && downcast<CSSPrimitiveValue>(fallback)->isString())
+                    return downcast<CSSPrimitiveValue>(fallback)->stringValue().impl();
+            }
+            return emptyAtom();
```

Comment:
This equivalent code is shorter:

```suggestion
            auto fallback = dynamicDowncast<CSSPrimitiveValue>(value->fallback());
            return fallback ? fallback.stringValue().impl() : emptyAtom();
```
```

Reply:
Thank much better indeed.

Reply:
@nt1m in this case I should also  verify if fallback is a string value, if not  display nothing.